lib/dao.js

```diff
@@ -52,6 +52,45 @@ function getIdValue(m, data) {
   return data && data[idName(m)];
 }
 
+function copyData(from, to) {
+  for (var key in from) {
+    to[key] = from[key];
+  }
+}
+
+function convertDataPropertiesByType(data) {
+  var typedData = {};
+  for (var key in data) {
+    // Convert the properties by type
+    typedData[key] = data[key];
+    if (typeof typedData[key] === 'object'
+            && typedData[key] !== null
+            && typeof typedData[key].toObject === 'function') {
+      typedData[key] = typedData[key].toObject();
+    }
+  }
+  return typedData;
+}
+
+DataAccessObject.prototype.strictChk = function strictChk(strict, data, inst, cb) {
```

Comment:
Don't use abbreviations please. 

Also if this is a private method, then it would be better to move it out of `DataAccessObject.prototype`. Otherwise it would end up mixed into Model classes, e.g. `Car.prototype.strictChk`.